packages/middleware-sdk-transcribe-streaming/src/websocket-handler.ts

```diff
@@ -22,98 +22,144 @@ export class WebSocketHandler implements HttpHandler {
     handlerProtocol: "websocket",
   };
   private readonly connectionTimeout: number;
+  private readonly sockets: Record<string, WebSocket[]>;
+
   constructor({ connectionTimeout }: WebSocketHandlerOptions = {}) {
     this.connectionTimeout = connectionTimeout || 2000;
+    this.sockets = {};
+  }
+
+  /**
+   * Destroys the WebSocketHandler.
+   * Closes all sockets from the socket pool.
+   */
+  destroy(): void {
+    for (const [key, sockets] of Object.entries(this.sockets)) {
+      for (const socket of sockets) {
+        socket.close(1000, `Socket closed through destroy() call`);
+      }
+      delete this.sockets[key];
+    }
   }
 
-  destroy(): void {}
+  /**
+   * Removes all closing/closed sockets from the socket pool for URL.
+   */
+  private removeNotUsableSockets(url: string): void {
```

Comment:
rename to `removeClosedSockets`?

Reply:
It removes Closing sockets too, that's why I'd renamed it from `Closed` to `NotUsable`.